template/addons/next-auth/protected-router.ts

```diff
@@ -0,0 +1,17 @@
+import * as trpc from "@trpc/server";
+import { createRouter } from "./context";
+
+export function createProtectedRouter() {
+  return createRouter().middleware(({ ctx, next }) => {
+    if (!ctx.session || !ctx.session.user) {
+      throw new trpc.TRPCError({ code: "UNAUTHORIZED" });
+    }
+    return next({
+      ctx: {
+        ...ctx,
```

Comment:
Didn't Alex say this wasn't needed or was that new in tRPC v10?

Reply:
Not sure about v10, yet to see the updates.

But Theo asked for this [here](https://github.com/t3-oss/create-t3-app/issues/191)

@juliusmarminge